crates/subspace-proof-of-time/Cargo.toml

```diff
@@ -0,0 +1,15 @@
+[package]
+name = "subspace-proof-of-time"
+description = "Subspace proof of time implementation"
+license = "Apache-2.0"
+version = "0.1.0"
+authors = ["Rahul Subramaniyam <rahulksnv@gmail.com>"]
+edition = "2021"
+include = [
+    "/src",
+    "/Cargo.toml",
+]
+
+[dependencies]
+aes = "0.8.3"
+subspace-core-primitives = { version = "0.1.0", path = "../subspace-core-primitives" }
```

Comment:
I don't think `thiserror` should be removed. It needs to be added and used for errors in `std` environment.

This crate wasn't and still isn't `no_std` compatible. Search `no_std` in the repo for examples of how we use `std` feature to achieve actual `no_std` compatibility in crates.

Reply:
Given different crates have different levels of support for `no_std`, or some don't have it at all (parking_lot vs spin for example), how do we check if something is `no_std` compatible? (e.g) is there a build target that would fail to compile for no_std, if a change breaks no_std compatibility?

Otherwise, this looks like an error prone process ..

Reply:
Well, we have numerous crates that are `no_std` compatible in the repo, this one is no different.

If crate will end up being included in the runtime, it must compile to WASM, which the way we compile it will fail to compile or run when crate doesn't support std. We can also create a CI action that will do these checks deliberately, but no one bothered to do that just yet.

Reply:
>  which the way we compile it will fail to compile or run when crate doesn't support std.

In this case, I don't see anything failing, to alert the no_std issue. Where do I look for the failure? Or how do I test to catch it?

Reply:
This crate is not used anywhere, why would it fail to compile in `no_std`? And it is absolutely clear it doesn't support `no_std`. To test you start with adding `#![no_std]` to the crate.

Reply:
Don't know what you mean, since it is part of the compilation. But the right solution would be have a no_std target that is part of CI/CD. But fixed these for now in this PR